### PR TITLE
[BC-Breaking] Remove lazy behavior from MelScale

### LINKED ISSUE #851: Blank zeroed rows in Mel spectrograms
## 🐛 Bug

Mel spectrogram have some random rows that are all zeros. Normal spectrograms are fine.

## To Reproduce

1. `mel = torchaudio.transforms.MelSpectrogram()(waveform)`

or

1. `spec = torchaudio.transforms.Spectrogram()(waveform)`
1. `mel = torchaudio.transforms.MelScale()(spec)`

Both give something like this (white lines are zero valued)

Screenshot 2020-08-04 at 07 24 27

## Expected behavior

All rows to have data!

## Environment

 - Tried both pytorch 1.5 and 1.6

### LINKED ISSUE #1569: Remove lazy behavior from MelScale
Address https://github.com/pytorch/audio/issues/1510

In 0.9.0 we are deprecating the lazy behavior of MelScale because it can make an invalid TorchScript object and it does not align with the design of torchaudio. Now in master branch, we can remove the implementation.

See [CONTRIBUTING.md](https://github.com/pytorch/audio/blob/master/CONTRIBUTING.md) for setting up the development environment.

## Steps
1.  Replace [`n_stft: Optional[int] = None`](https://github.com/pytorch/audio/blob/6f9c14bec787d952c2aa1b6a13d11509c7a38bed/torchaudio/transforms.py#L258) with `n_stft: int = 201`. 
2. Remove [the branches](https://github.com/pytorch/audio/blob/6f9c14bec787d952c2aa1b6a13d11509c7a38bed/torchaudio/transforms.py#L271-L282) for `n_stft = None`
3. Remove [`__prepare_script__`](https://github.com/pytorch/audio/blob/6f9c14bec787d952c2aa1b6a13d11509c7a38bed/torchaudio/transforms.py#L285-L297) method
4. Remove [the case `numel() == 0`](https://github.com/pytorch/audio/blob/6f9c14bec787d952c2aa1b6a13d11509c7a38bed/torchaudio/transforms.py#L312-L319)
5. Remove [test](https://github.com/pytorch/audio/blob/6f9c14bec787d952c2aa1b6a13d11509c7a38bed/test/torchaudio_unittest/transforms/torchscript_consistency_impl.py#L62-L65)

## Tests

Run test `cd test && pytest torchaudio_unittest.transforms -vk MelScale` and make changes to failures.

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #1569
Closes #851

### CODE REVIEW

**Comment by Patil2099** [comment]
I will make changes to the failures soon.

File: torchaudio/transforms.py [hunk 1]
```diff
@@ -268,33 +268,8 @@ def __init__(self,
...
-                'Initialization of torchaudio.transforms.MelScale with an unset weight '
-                '`n_stft=None` is deprecated and will be removed in release 0.10. '
-                'Please set a proper `n_stft` value. Typically this is `n_fft // 2 + 1`. '
-                'Refer to https://github.com/pytorch/audio/issues/1510 '
-                'for more details.'
-            )
-
-        fb = torch.empty(0) if n_stft is None else F.create_fb_matrix(
-            n_stft, self.f_min, self.f_max, self.n_mels, self.sample_rate, self.norm,
-            self.mel_scale)
```
Comment (mthrok) [inline torchaudio/transforms.py:282]: `fb = F.create_fb_matrix(...` needs to be retrained.

**Review by mthrok — COMMENTED** [review]
Hi @Patil2099

Thanks for working on this. Please refer to my comment for the fix.

File: torchaudio/transforms.py [hunk 2]
```diff
@@ -267,34 +267,11 @@ def __init__(self,
...
-            warnings.warn(
-                'Initialization of torchaudio.transforms.MelScale with an unset weight '
-                '`n_stft=None` is deprecated and will be removed in release 0.10. '
-                'Please set a proper `n_stft` value. Typically this is `n_fft // 2 + 1`. '
-                'Refer to https://github.com/pytorch/audio/issues/1510 '
-                'for more details.'
-            )
-
-        fb = torch.empty(0) if n_stft is None else F.create_fb_matrix(
+        F.create_fb_matrix(
```
Comment (mthrok) [inline torchaudio/transforms.py:270]:
```suggestion
        fb = F.create_fb_matrix(
```

**Comment by mthrok** [comment]
Addressed in #1636 with credit carried out. Thanks!

**Pull request closed without merging**